File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    bool isVowel(char c){
            return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || 
               c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U';
    }
    
    string reverseVowels(string s) {
        int i = 0, j = s.size() - 1;
        
        while(i < j){
            if(!isVowel(s[i])){
                i++;
                continue;
            }
            if(!isVowel(s[j])){
                j--;
                continue;
            }
            swap(s[i++], s[j--]);
        }
        return s;
    }
    
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    let str = s.split("");
    
    let left = 0;
    let right = str.length-1;
    
    const isVowel = (a) => {
        return "aeiouAEIOU".includes(a);
    }
    
    while(left < right){
        if(isVowel(str[left]) && isVowel(str[right])){
            [str[left], str[right]] = [str[right], str[left]];
            left++;
            right--;
        } else if(isVowel(str[left]) && !isVowel(str[right])){
            right--;
        } else if(!isVowel(str[left]) && isVowel(str[right])){
            left++;
        } else{
            left++;
            right--;
        }
    }
    
    return str.join("");
};